syntax = "proto3";

service TutoringService {
    // RPC to get a response from the tutoring LLM
    rpc GetTutoringResponse (TutoringRequest) returns (TutoringResponse);
}

message TutoringRequest {
    string course_name = 1;
    string query = 2;
    string auth_token = 3;  // Add an auth token for server-to-server verification
}


message TutoringResponse {
    string response = 1;
}
